jupyter_notebooks/Analyzing public and hybrid cloud networks .ipynb

```diff
@@ -0,0 +1,567 @@
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Initialize a network and snapshot\n",
+    "NETWORK_NAME = \"hybrid-cloud\"\n",
+    "SNAPSHOT_NAME = \"snapshot\"\n",
+    "\n",
+    "SNAPSHOT_PATH = \"networks/hybrid-cloud\"\n",
+    "\n",
+    "bf_set_network(NETWORK_NAME)\n",
+    "bf_init_snapshot(SNAPSHOT_PATH, name=SNAPSHOT_NAME, overwrite=True, extra_args={ \"parsereuse\" : False})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The network snapshot that we just initialized is illustrated below. It has a datacenter network with the standard leaf-spine design on the left. Though not strictly necessary, we have included a host `srv-101` in this network to enable end-to-end analysis. The exit gateway of the datacenter connects to an Internet service provider that we call `isp_dc`.\n",
+    "\n",
+    "The AWS network is shown on the right. It is spread across two regions, `us-east-2` and `us-west-2`. Each region has two VPCs, one of which is meant to host Internet-facing services and the other is meant to host only private services. Subnets in the public-facing VPCs use an Internet gateway to send and receive traffic outside of AWS. The two VPCs in a region peer via a transit gateway. Each VPC has two subnets, and we have some instances running as well. \n",
+    "\n",
+    "The physical network connects to the AWS network by having the exitgw establish IPSec tunnels, shown in pink, to the two transit gateways. BGP sessions running atop these tunnels to make endpoints aware of prefixes on the other side.\n",
+    "\n",
+    "You can view configuration files that we used [here](https://github.com/batfish/pybatfish/tree/master/jupyter_notebooks/networks/hybrid-cloud). The AWS portion of the configuration is in the aws_configs subfolder. It has JSON files obtained via AWS APIs. An example script that packages AWS data into a Batfish snapshot is [here](https://github.com/ratulm/bf-aws-snapshot).\n",
+    "\n",
+    "![example-bgp-network](https://raw.githubusercontent.com/batfish/pybatfish/aws-notebook/jupyter_notebooks/networks/hybrid-cloud/hybrid-cloud.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Analyzing network paths\n",
+    "\n",
+    "Batfish can help analyze cloud and hybrid networks by showing how exactly traffic flows (or not) in the network, which can help debug and fix configuration errors. Batfish can also help ensure that the network is configured exactly as desired, with respect to reachability and security policies. We illustrate these types of analysis below.\n",
+    "\n",
+    "First, lets define a couple of maps to help with the analysis."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Instances in AWS in each region and VPC type (public, private)\n",
```

Comment:
is there a way to get the mapping from the AWS JSON data itself?

Reply:
probably, but we are not parsing the AWS JSON here.